session/handler.go

```diff
@@ -459,6 +485,92 @@ func (h *Handler) IsAuthenticated(wrap httprouter.Handle, onUnauthenticated http
 	}
 }
 
+// swagger:parameters adminSessionRefresh
+// nolint:deadcode,unused
+type adminSessionRefresh struct {
+	// ID is the session's ID.
+	//
+	// required: true
+	// in: path
+	ID string `json:"id"`
+}
+
+// swagger:route PATCH /sessions/refresh/{id} v0alpha2 adminSessionRefresh
+//
+// Calling this endpoint refreshes a given session.
+// If `session.earliest_refresh` is set it will only refresh the session after this time has passed.
+//
+// This endpoint is useful for:
+//
+// - Session refresh
+//
+//     Schemes: http, https
+//
+//     Security:
+//       oryAccessToken:
+//
+//     Responses:
+//       200: session
+//       404: jsonError
+//       500: jsonError
+func (h *Handler) adminSessionRefresh(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	iID, err := uuid.FromString(ps.ByName("id"))
+	if err != nil {
+		h.r.Writer().WriteError(w, r, herodot.ErrBadRequest.WithError(err.Error()).WithDebug("could not parse UUID"))
+		return
+	}
+	s, err := h.r.SessionPersister().GetSession(r.Context(), iID)
+	if err != nil {
+		h.r.Writer().WriteError(w, r, err)
+		return
+	}
+	c := h.r.Config(r.Context())
+	if s.CanBeRefreshed(c) {
+		if err := h.r.SessionPersister().UpsertSession(r.Context(), s.Refresh(c)); err != nil {
+			h.r.Writer().WriteError(w, r, err)
+			return
+		}
+	}
+
+	h.r.Writer().Write(w, r, s)
+}
+
+// swagger:route PATCH /sessions/refresh v0alpha2 adminCurrentSessionRefresh
+//
+// Calling this endpoint refreshes a given session.
+// If `session.earliest_refresh` is set it will only refresh the session after this time has passed.
+//
+// This endpoint is useful for:
+//
+// - Session refresh
+//
+//     Schemes: http, https
+//
+//     Security:
+//       oryAccessToken:
+//
+//     Responses:
```

Comment:
I think this route would currently cause a conflict with the other session routes? I would suggest:

```suggestion
// swagger:route PATCH /sessions?refresh=true v0alpha2 adminRefreshSessionFromCredentials
//
// Calling this endpoint you can refresh an Ory Session Cookie or Ory Session Token. You must
// include the Ory Session Token in the `Cookie` header or the Ory Session Token in the `X-Session-Token`
// header.
//
// If `session.earliest_refresh` is set it will only refresh the session after this time has passed.
//
//     Schemes: http, https
//
//     Security:
//       oryAccessToken:
//
//     Responses:
```

Please also add these keys (x-session-token, cookie) to the argument this method can receive (see the whoami endpoint for an example) :)